src/ts/components/Icon.ts

```diff
@@ -1,35 +1,22 @@
 import { Component, ComponentConfig } from './Component';
-import { i18n, LocalizableText } from '../localization/i18n';
 import { DOM } from '../DOM';
 
-export interface IconConfig extends ComponentConfig {
-  /**
-   * WCAG20 standard for defining info about the component (usually the name)
-   */
-  ariaLabel?: LocalizableText;
-  /**
-   * Alternative text for the icon, used for accessibility (e.g., screen readers).
-   */
-  altText?: LocalizableText;
-}
-
-export class Icon extends Component<IconConfig> {
-  constructor(config: IconConfig) {
+export class Icon extends Component<ComponentConfig> {
+  constructor(config: ComponentConfig) {
```

Comment:
Nit: I was thinking if we should set a default value here to not need the empty object on the caller side 🤔 


```suggestion
  constructor(config: ComponentConfig = {}) {
```

Reply:
makes sense, i'll clean it up!